zerver/lib/test_fixtures.py

```diff
@@ -144,6 +176,12 @@ def is_template_database_current(
                                     for setting_name in check_settings])
         hash_status = files_hash_status and settings_hash_status
 
-        return are_migrations_the_same(migration_status, settings=settings) and hash_status
+        if not hash_status:
+            return 1
+
+        if not are_migrations_the_same(migration_status, settings=settings):
+            return 2
```

Comment:
We need better support logic here -- we can't reverse migrations, so we need to distinguish the case where it's a subset.

Reply:
I didn't really understand this one.

Reply:
So, if you applied some migrations from one branch, and then switch to another, that first branch's migrations Python code no longer exist in your tree, so you can't run the reversal code for them.  Thus, we have to restart from scratch.

Probably worth adding a comment explaining this :)

Reply:
oh yeah, I missed it totally. 
So when I thought about just using the difference between the migration output, I initially thought about problems with deleting but then thought since we don't delete migrations in master that won't be a problem totally missing out the fact that switching branches can cause a similar issue.